### ISSUE TITLE
Cannot use Group as Table name && cannot autoSchemeSync when use alias Entity

### ISSUE THREAD

**Issue by muyu66** [typeorm/typeorm#423]
Hello, i may found a bug. First we can not use 'group' as Table name, because it can throw error in autoSchemeSync.

Then, i code

```
@Index('Groups name', ['name'], { unique: true })
@Entity('groups')
class Group {
```

But there was something strange. When in autoSchemeSync, it throw

```
Error: ER_CANT_DROP_FIELD_OR_KEY: Can't DROP 'name'; check that
column/key exists
```

I think it is a problem of Entity alias table name or Entity Model name.

**If you need, i will provide more details**

**Comment by valVk** [comment]
https://github.com/typeorm/typeorm/issues/334 - probably escaping problems but i may be wrong

**Comment by pleerock** [comment]
no I don't think so, probably bug in index creation

**Comment by muyu66** [comment]
@pleerock I think this index create a binding of Group and 'name' field rather than the binding of real table name and 'name' field.

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.23`. You can try it via `npm i typeorm@next`.
